cdap-app-fabric/src/main/java/co/cask/cdap/scheduler/NotificationSubscriberService.java

```diff
@@ -313,21 +267,40 @@ protected void updateJobQueue(DatasetContext context, Notification notification)
       }
       DatasetId datasetId = DatasetId.fromString(datasetIdString);
       for (ProgramSchedule schedule : getSchedules(context, Schedulers.triggerKeyForPartition(datasetId))) {
-        addJob(new Job(schedule));
+        // long publishTimestamp = new MessageId(Bytes.fromHexString(message.getId())).getPublishTimestamp();
+        // should we use subscribe time (System.currentTimeMillis?)
+
+        JobQueueDataset jobQueue = getJobQueue(context);
+        boolean existingJobModified = false;
+        try (CloseableIterator<Job> jobs = jobQueue.getJobsForSchedule(schedule.getScheduleId())) {
+          while (jobs.hasNext()) {
```

Comment:
does the job queue somehow make sure that there is at most one job per schedule that is not pending launch? Otherwise you would create multiple jobs here.